### ISSUE TITLE
CDFT Becke weight function breaks for small ATOMIC_RADII

### ISSUE THREAD

**Issue by chrisahart** [cp2k/cp2k#850]
Dear CP2K developers,

I have found that when ionic radii are used for a bulk MgO crystal the Becke weight function appears to break ([see image](https://user-images.githubusercontent.com/10100029/78054986-c9f98a00-737a-11ea-9e08-174ac58f349b.png)). When using covalent radii however the weight function is calculated correctly ([see image](https://user-images.githubusercontent.com/10100029/78055109-f3b2b100-737a-11ea-9a4b-1c332c702852.png)).

More generally, it appears that for certain systems the Becke weight function breaks when the constrained atoms have ATOMIC_RADII smaller than the unconstrained atoms. I have confirmed this behaviour for two different structures on two different machines using CP2K 7.1.

I include an example input and output file for the system with ionic and covalent radii.

Ionic radii: Mg 0.72 Å, O 1.40 Å  (Shannon, R. D. (1976))
[Input](https://github.com/cp2k/cp2k/files/4410567/ionic_input.txt)
[Output](https://github.com/cp2k/cp2k/files/4410568/ionic_output.txt)
[Geometry](https://github.com/cp2k/cp2k/files/4410583/input.txt)

Covalent radii: Mg 1.39 Å, O 0.63 Å (Pyykkö, P., & Atsumi, M. (2009)
[Input](https://github.com/cp2k/cp2k/files/4410574/covalent_input.txt)
[Output](https://github.com/cp2k/cp2k/files/4410575/covalent_output.txt)

(Originally posted on CP2K user forum, moved here as I believe this to be a bug: https://groups.google.com/forum/#!searchin/cp2k/ahart|sort:date/cp2k/1yTNsYN6nRQ/S9HSmHXBAQAJ)

**Comment by oschuett** [comment]
This was presumably fixed by #2111.

**Comment by chrisahart** [comment]
> This was presumably fixed by #2111.

I do not believe so. #2111 is a change to the Hirshfeld partitioning CDFT, not to the Becke partitioning CDFT which is the cause of this reported bug.

**Comment by oschuett** [comment]
My bad, thanks for clarifying.

**Comment by DCM-Uni-Paderborn** [comment]
This is directly addressed by #5245.

The issue was reproduced as a cube output formatting problem for very small Becke weight values: values with three-digit exponents could be written without an explicit `E`, e.g. `0.33004-101`, which some cube viewers misparse and display as artificial isosurfaces. #5245 switches cube output to an explicit three-digit exponent format while preserving the 13-character field width, and adds a small regression test for this case.

Closing this as fixed by #5245.
